src/Provider/ProductUrlProvider.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Menu plugin for Sylius.
+ *
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusMenuPlugin\Provider;
+
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class ProductUrlProvider extends AbstractUrlProvider
+{
+    public const PROVIDER_CODE = 'product';
+
+    protected string $code = self::PROVIDER_CODE;
+
+    protected string $icon = 'cube';
+
+    protected int $priority = 50;
+
+    public function __construct(
+        RouterInterface $router,
+        private ProductRepositoryInterface $productRepository,
+    ) {
+        parent::__construct($router);
+    }
+
+    public function getItems(string $locale): array
+    {
+        $products = $this->productRepository->findBy(['enabled' => true]);
```

Comment:
I think it can be done as improvement later, if we make a new release as fast as we can.
Actually it is a simple array with name and value in results, so even if you have like 1.000 results, you will have a long scroll in modal but the feature can start to be used.

WDYTOT ?

Reply:
I added a search !